### PR TITLE
EL-26: Refactor ConfigModule & ConfigService to Match NestJS Pattern

### PR DESCRIPTION
**Pull request by yachujoshi-lf**
Title: Refactor ConfigModule & ConfigService to Match NestJS Pattern.

Description:
This PR updates the existing setup to add custom `ConfigModule` & `ConfigService` to pull out `env` variables.

Jira:
https://lftechnology.atlassian.net/browse/EL-26

Summary of Changes:
[x] Create custom ConfigModule and ConfigService
[x] Register env variables into their own scope.
[x] Implement utility functions to initialize module factory.
[x] Remove dead code from the application.
[x] Change `Container` interface props to match the type.

### CODE REVIEW

File: apps/server/src/config/config.service.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    const value = this.configService.get<T>(key);
+
+    if (!value) {
+      throw new Error(`Config for ${key} os undefined`);
+    }
+
+    return value;
+  }
+
+  getAppConfig() {
```
Comment (SamipSthaLF) [inline apps/server/src/config/config.service.ts:18]: add code doc.

File: apps/server/src/config/db.config.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { registerAs } from '@nestjs/config';
+
+import { DB } from '@/constants/key.constant';
+
+export const dbConfig = registerAs(DB, () => ({
```
Comment (SamipSthaLF) [inline apps/server/src/config/db.config.ts:5]: .

File: apps/server/src/config/google-strategy.config.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { registerAs } from '@nestjs/config';
+
+import { GOOGLE_STRATEGY } from '@/constants/key.constant';
+
+export const googleStrategyConfig = registerAs(GOOGLE_STRATEGY, () => ({
```
Comment (SamipSthaLF) [inline apps/server/src/config/google-strategy.config.ts:5]: .

**Review by alinagoal — APPROVED** [review]

**Review by SamipSthaLF — APPROVED** [review]

**Comment by SamipSthaLF** [comment]
Good to go. 👍

**Pull request merged**